pytorch_lightning/tuner/tuning.py

```diff
@@ -112,22 +149,33 @@ def scale_batch_size(
                 - ``model.hparams``
                 - ``trainer.datamodule`` (the datamodule passed to the tune method)
         """
-        self.trainer.auto_scale_batch_size = True
-        result = self.trainer.tune(
-            model,
-            train_dataloaders=train_dataloaders,
-            val_dataloaders=val_dataloaders,
-            datamodule=datamodule,
-            scale_batch_size_kwargs={
-                "mode": mode,
-                "steps_per_trial": steps_per_trial,
-                "init_val": init_val,
-                "max_trials": max_trials,
-                "batch_arg_name": batch_arg_name,
-            },
+        self.trainer.state.fn = TrainerFn.TUNING
+        self.tuning = True
+
+        _check_tuner_configuration(self.trainer, train_dataloaders, val_dataloaders, dataloaders, method)
+
+        batch_size_finder = BatchSizeFinder(
+            mode=mode,
+            steps_per_trial=steps_per_trial,
+            init_val=init_val,
+            max_trials=max_trials,
+            batch_arg_name=batch_arg_name,
         )
+        batch_size_finder._early_exit = True
+        self.trainer.callbacks = [batch_size_finder] + self.trainer.callbacks
```

Comment:
Verify the `BatchSizeFinder` doesn't exist already within the callbacks.

Reply:
already there :)